backend/LexBoxApi/Services/HgService.cs

```diff
@@ -289,23 +289,25 @@ public Task<HttpContent> InvalidateDirCache(string code)
             }
             catch (Exception) { }
         }
-        var result = ExecuteHgCommandServerCommand(code, "invalidatedircache", default);
+        var result = ExecuteHgCommandServerCommand(code, "invalidatedircache", token);
         return result;
     }
 
-    public async Task<string> GetTipHash(string code)
+    public async Task<string> GetTipHash(string code, CancellationToken token = default)
     {
-        var content = await ExecuteHgCommandServerCommand(code, "tip", default);
+        var content = await ExecuteHgCommandServerCommand(code, "tip", token);
         return await content.ReadAsStringAsync();
     }
 
-    public async Task WaitForRepoEmptyState(string code, RepoEmptyState expectedState)
+    private async Task WaitForRepoEmptyState(string code, RepoEmptyState expectedState, int timeoutMs = 30_000, CancellationToken token = default)
     {
-        // TODO: Set timeout so uploading a zip of an empty .hg repo doesn't cause infinite loop here
+        // Set timeout so unforeseen errors can't cause an infinite loop
+        var timeoutSource = CancellationTokenSource.CreateLinkedTokenSource(token);
+        timeoutSource.CancelAfter(30_000);
```

Comment:
you probably meant to use the timeoutMs parameter here instead of hardcoding the timeout

Reply:
Yes, I did. Fixed now, thanks for the catch.

Reply:
Done in commit 18ae58e.